contributors/design-proposals/storage/grow-flexvolume-size.md

```diff
@@ -0,0 +1,170 @@
+    return true
+  }
+  ...
+}
+```
+
+### FlexVolume Plugin Changes
+
+FlexVolume relies on underneath volume driver to implement various volume functions, e.g. attach/detach. As a result, volume driver will decide whether volume can be expanded or not. 
+
+By default, we assume all kinds of flex volume drivers support resizing. If they do not, flex volume plugin can detect this during resizing call to flex volume driver and always throw out error to stop the resizing process. So as a result, to implement resizing feature in flex volume plugin, the plugin itself must implement the following `ExpandableVolumePlugin` interfaces:
+
+#### ExpandVolumeDevice
+
+Volume resizing controller invokes this method while receiving a valid PVC resizing request. FlexVolume plugin calls the underneath volume driver’s corresponding `expandvolume` method with three parameters, including new size of volume(number in bytes), old size of volume(number in bytes) and volume spec, to expand PV. Once the expansion is done, volume driver should return the new size(number in bytes) of the volume to FlexVolume.
+
+A sample implementation of `ExpandVolumeDevice` method is like:
+```
+func (plugin *flexVolumePlugin) ExpandVolumeDevice(spec *volume.Spec, newSize resource.Quantity, oldSize resource.Quantity) (resource.Quantity, error) {
+  const timeout = 10*time.Minute
+ 
+  call := plugin.NewDriverCallWithTimeout(expandVolumeCmd, timeout)
+  call.Append(newSize.Value())
+  call.Append(oldSize.Value())
+  call.AppendSpec(spec, plugin.host, nil)
+
+  // If the volume driver does not support resizing, Flex Volume Plugin can throw out error here
+  // to stop expand controller's resizing process. 
+  ds, err := call.Run()
+  if err != nil {
+      return resource.NewQuantity(0, resource.BinarySI), err
+  }
+ 
+  return  resource.NewQuantity(ds.ActualVolumeSize, resource.BinarySI), nil
+}
+```
+
+Add a new field in type `DriverStatus` named `ActualVolumeSize` to identify the new expanded size of the volume returned by underneath volume driver:
+```
+// DriverStatus represents the return value of the driver callout.
+type DriverStatus struct {
+  ...
+  ActualVolumeSize int64 `json:"volumeNewSize,omitempty"`
+}
+```
+
+#### RequiresFSResize
+
+`RequiresFSResize` is a method to implement `ExpandableVolumePlugin` interface. The return value of this method identifies whether or not a file system resize is required once physical volume get expanded. If the return value is `true`, PV resize controller will consider the volume resize operation is done and then update the PV object’s capacity in K8s directly; If the return value is `false`, PV resize controller will leave kubelet to do the file system resize, and kubelet on worker node will call `ExpandFS` method of FlexVolume to finish the file system resize step(at present, only offline FS resize is supportted, online resize support is under community discussion [here](https://github.com/kubernetes/community/pull/1535)). 
```

Comment:
"If the return value is **`false`**, PV resize controller will consider the volume resize operation is done..."

Reply:
That's my mistake, need to exchange the behavior of the two return values. I'll update this in the next patch update.